apps/avatax/src/pages/api/webhooks/order-confirmed.ts

```diff
@@ -72,23 +73,37 @@ export default wrapWithLoggerContext(
         logger.debug("Confirming order...");
 
         if (taxProviderResult.isOk()) {
-          const confirmedOrder = await taxProviderResult.value.confirmOrder(payload.order);
-
-          logger.info("Order confirmed", { confirmedOrder });
-          const client = createInstrumentedGraphqlClient({
-            saleorApiUrl,
-            token,
-          });
-
-          const orderMetadataManager = new OrderMetadataManager(client);
-
-          await orderMetadataManager.updateOrderMetadataWithExternalId(
-            payload.order.id,
-            confirmedOrder.id,
-          );
-          logger.info("Updated order metadata with externalId");
+          try {
+            const confirmedOrder = await taxProviderResult.value.confirmOrder(payload.order);
+
+            logger.info("Order confirmed", { confirmedOrder });
+            const client = createInstrumentedGraphqlClient({
+              saleorApiUrl,
+              token,
+            });
+
+            const orderMetadataManager = new OrderMetadataManager(client);
+
+            await orderMetadataManager.updateOrderMetadataWithExternalId(
+              payload.order.id,
+              confirmedOrder.id,
+            );
+            logger.info("Updated order metadata with externalId");
+
+            return webhookResponse.success();
+          } catch (error) {
+            logger.debug("Error confirming order", { error });
+
+            switch (true) {
+              case error instanceof TaxBadPayloadError: {
+                // @ts-ignore - Vercel uses old version of Typescript
+                return res.status(400).send(error.message);
+              }
+            }
 
-          return webhookResponse.success();
+            logger.error("Unhandled error executing webhook", { error });
+            return webhookResponse.error(error);
```

Comment:
question: how about we use a plain `if` instead of `switch(true)`? tbh it looks a bit odd to me

Reply:
I used 114 line as inspiration 😄 I imagine in the future  there may be more of these errors that we want to handle

Reply:
@witoszekdev I haven't yet found a good syntax for such a pattern matching. Using switch is close to pattern matching from functional languages. We can implement this with "ifs" but you will also see problems to write this:

```
case Error.Y:
case Error.X:
case Error.Z: {
  return ...
} 
```

"If" will be ugly too.

I suggest we don't bother about this now and if someone has an idea to write a good (and typesafe if possible) syntax, we will refactor every place

eg https://github.com/gvergnaud/ts-pattern

Reply:
Oh, sorry I didn't see what was below 🙈